hazelcast/src/main/java/com/hazelcast/nio/tcp/AbstractSelectionHandler.java

```diff
@@ -93,4 +112,47 @@ final void registerOp(int operation) {
         }
     }
 
+    final void unregisterOp(int operation) {
+        SelectionKey selectionKey = getSelectionKey();
+        try {
+            selectionKey.interestOps(selectionKey.interestOps() & ~operation);
+        } catch (Throwable e) {
+            handleSocketException(e);
+        }
+    }
+
+    public IOSelector getOwner() {
+        return ioSelector;
+    }
+
+    private class MigrationTask implements Runnable {
+        private final IOSelector newOwner;
+
+        public MigrationTask(IOSelector newOwner) {
+            this.newOwner = newOwner;
+        }
+
+        @Override
+        public void run() {
+            if (!socketChannel.isOpen()) {
+                return;
+            }
+            unregisterOp(initialOps);
+            ioSelector = newOwner;
```

Comment:
Just did a quick scan, it seems`MigrationTask` is executed by IO thread itself. So, no need.